service/matching/matching_engine.go

```diff
@@ -134,8 +142,7 @@ type (
 		metricsHandler       metrics.Handler
 		partitionsLock       sync.RWMutex // locks mutation of partitions
 		partitions           map[tqid.PartitionKey]taskQueuePartitionManager
-		taskQueueCountLock   sync.Mutex
-		taskQueueCount       map[taskQueueCounterKey]int // per-namespace task queue counter
+		gaugeMetricCounter   gaugeMetrics // per-namespace task queue counter
```

Comment:
for this kind of thing, where you have a single variable/field of a type, it's nice to name them the same:
```suggestion
		gaugeMetrics   gaugeMetrics // per-namespace task queue counters
```